lib/extensions/method.rb

```diff
@@ -130,13 +136,19 @@ def invalid_named_args?
         pair.first == :keyreq
       end
 
+      unlimited = named_params.any? do |pair|
+        pair.first == :keyrest
+      end
+
```

Comment:
Check the source method to see if it takes unbounded named arguments (**named_args), if it does then any number of named arguments are allowed here as long as the required named arguments have been supplied.